src/renderer/pages/Onboarding/Vault/ManageDynamicDerivations/ManageDynamicDerivations.tsx

```diff
@@ -60,11 +65,37 @@ export const ManageDynamicDerivations = ({ seedInfo, onBack, onComplete }: Props
   }, [onComplete]);
 
   useEffect(() => {
-    const chains = chainsService.getChainsData();
-    const chainsMap = keyBy(chainsService.sortChains(chains), 'chainId');
-    setChainsObject(chainsMap);
+    const chainIds = chainsService.getChainsData({ sort: true }).map((chain) => chain.chainId);
+    setChainsIds(chainIds);
   }, []);
 
+  useEffect(() => {
+    if (chainsIds.length === 0) return;
+
+    const chainsMap = chainsIds.reduce<Record<ChainId, Array<ChainAccount | ShardAccount[]>>>((acc, chainId) => {
+      return { ...acc, [chainId]: [] };
+    }, {});
```

Comment:
Why do you use `reduce` here? You can use `keyBy` from lodash, or dictionary function from `shared/lib/utils`